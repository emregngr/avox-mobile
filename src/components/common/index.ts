export * from './FaqItem'
export * from './FormElements'
export * from './FullScreenLoading'
export * from './Header'
export * from './OnboardingItem'
export * from './ProfileItem'
export * from './RenderDetailTabBar'
export * from './RenderTabBar'
export * from './SafeLayout'
export * from './SearchInput'
export * from './ThemedButton'
export * from './ThemedButtonText'
export * from './ThemedGradientButton'
export * from './ThemedText'
export * from './ToastConfig'
