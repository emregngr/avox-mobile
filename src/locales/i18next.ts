import { getLocales } from 'expo-localization'
import i18next from 'i18next'
import { getI18n, initReactI18next } from 'react-i18next'

import en from '@/locales/en.json'
import tr from '@/locales/tr.json'

const locales = getLocales()
i18next.language = locales[0]?.languageTag ?? 'en'

export const languageResources = i18next.use(initReactI18next).init({
  fallbackLng: 'en',
  resources: {
    en: { translation: en },
    tr: { translation: tr },
  },
})

export const getLocale = (key: string, params?: Record<string, any>) => {
  const i18n = getI18n()
  let translatedText = i18n?.t(key)

  if (params) {
    Object.keys(params).forEach(paramKey => {
      translatedText = translatedText.replace(`{${paramKey}}`, params[paramKey])
    })
  }

  return translatedText
}

export const i18nChangeLocale = async (lng: string) => {
  const i18n = getI18n()
  await i18n?.changeLanguage(lng)
}

export default i18next
