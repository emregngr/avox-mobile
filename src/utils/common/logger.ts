import { getCrashlytics, log, recordError } from '@react-native-firebase/crashlytics'

export type LogCategory = 'info' | 'error' | 'warning' | 'debug'

export type LogData = Record<string, unknown> | Error | null

const crashlytics = getCrashlytics()

export const Logger = {
  breadcrumb: (message: string, category: LogCategory, data: LogData = null): void => {
    const row = {
      category,
      data: data ?? {},
      message,
    }

    if (!__DEV__) {
      if (data instanceof Error) {
        recordError(crashlytics, data)
      } else {
        log(crashlytics, JSON.stringify(row))
      }
    }

    console.log(message, category, data)
  },

  log: (message: string, data: LogData = null): void => {
    const row = {
      category: 'info' as LogCategory,
      data: data ?? {},
      message,
    }

    if (!__DEV__) {
      if (data instanceof Error) {
        recordError(crashlytics, data)
      } else {
        log(crashlytics, JSON.stringify(row))
      }
    }

    console.log(message, data)
  },
}
