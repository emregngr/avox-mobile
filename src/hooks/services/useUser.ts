import { getApp } from '@react-native-firebase/app'
import { getAuth } from '@react-native-firebase/auth'
import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
import { router } from 'expo-router'
import { Alert } from 'react-native'
import Toast from 'react-native-toast-message'

import { getLocale } from '@/locales/i18next'
import {
  addUserPassword,
  changeUserPassword,
  deleteUserAccount,
  getUser,
  registerDevice,
  registerDeviceToUser,
  updateUser,
} from '@/services/userService'
import { setIsAuthenticated } from '@/store/auth'
import useThemeStore from '@/store/theme'
import { deleteUser } from '@/store/user'
import type { AddPasswordCredentials, ChangePasswordCredentials } from '@/types/feature/password'
import type { ProfileData } from '@/types/feature/user'

const app = getApp()
const auth = getAuth(app)

export const useGetUser = () => {
  const userId = auth?.currentUser?.uid
  return useQuery({
    enabled: !!userId,
    queryFn: getUser,
    queryKey: ['userProfile', userId],
  })
}

export const useUpdateUser = () => {
  const queryClient = useQueryClient()
  const userId = auth?.currentUser?.uid
  const { selectedTheme } = useThemeStore()

  return useMutation<any, Error, ProfileData, { previousProfile: any }>({
    mutationFn: async (profileData: ProfileData) => await updateUser(profileData),
    onError: (error, newProfileData, context) => {
      queryClient.setQueryData(['userProfile', userId], context?.previousProfile)
      Alert.alert(
        getLocale('error'),
        getLocale('profileUpdateFailed'),
        [
          {
            text: getLocale('ok'),
          },
        ],
        {
          userInterfaceStyle: selectedTheme,
        },
      )
    },
    onMutate: async newProfileData => {
      await queryClient.cancelQueries({ queryKey: ['userProfile', userId] })
      const previousProfile = queryClient.getQueryData(['userProfile', userId])
      queryClient.setQueryData(['userProfile', userId], (oldData: any) => ({
        ...oldData,
        ...newProfileData,
        displayName: `${newProfileData?.firstName} ${newProfileData?.lastName}`,
      }))
      return { previousProfile }
    },
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey: ['userProfile', userId] })
    },
    onSuccess: () => {
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('profileUpdatedSuccessfully'),
        type: 'success',
      })
      router.back()
    },
  })
}

export const useChangePassword = () => {
  const { selectedTheme } = useThemeStore()

  return useMutation({
    mutationFn: async (credentials: ChangePasswordCredentials) =>
      await changeUserPassword(credentials),
    onError: () => {
      Alert.alert(
        getLocale('error'),
        getLocale('passwordChangeFailed'),
        [
          {
            text: getLocale('ok'),
          },
        ],
        {
          userInterfaceStyle: selectedTheme,
        },
      )
    },
    onSuccess: () => {
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('passwordChangedSuccessfully'),
        type: 'success',
      })
      router.back()
    },
  })
}

export const useAddPassword = () => {
  const { selectedTheme } = useThemeStore()

  return useMutation({
    mutationFn: async (credentials: AddPasswordCredentials) => await addUserPassword(credentials),
    onError: () => {
      Alert.alert(
        getLocale('error'),
        getLocale('passwordAddFailed'),
        [
          {
            text: getLocale('ok'),
          },
        ],
        {
          userInterfaceStyle: selectedTheme,
        },
      )
    },
    onSuccess: () => {
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('passwordAddedSuccessfully'),
        type: 'success',
      })
      router.back()
    },
  })
}

export const useDeleteUser = () => {
  const queryClient = useQueryClient()

  return useMutation({
    mutationFn: deleteUserAccount,
    onError: () => {
      Alert.alert(getLocale('error'), getLocale('accountDeleteFailed'), [
        {
          text: getLocale('ok'),
        },
      ])
    },
    onSuccess: async () => {
      await deleteUser()
      setIsAuthenticated(false)
      queryClient.clear()
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('accountDeletedSuccessfully'),
        type: 'success',
      })
    },
  })
}

export const useRegisterDevice = () =>
  useMutation({
    mutationFn: async () => await registerDevice(),
  })

export const useRegisterDeviceToUser = () =>
  useMutation({
    mutationFn: async () => await registerDeviceToUser(),
  })
