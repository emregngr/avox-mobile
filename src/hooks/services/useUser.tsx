import firebase from '@react-native-firebase/app'
import { getAuth } from '@react-native-firebase/auth'
import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
import { router } from 'expo-router'
import { Alert } from 'react-native'
import Toast from 'react-native-toast-message'

import { getLocale } from '@/locales/i18next'
import {
  addUserPassword,
  changeUserPassword,
  deleteUserAccount,
  getUser,
  registerDevice,
  registerDeviceToUser,
  updateUser,
} from '@/services/userService'
import { setIsAuthenticated } from '@/store/auth'
import { deleteUser } from '@/store/user'
import type { AddPasswordCredentials, ChangePasswordCredentials } from '@/types/feature/password'
import type { ProfileData } from '@/types/feature/user'

const auth = getAuth(firebase.app())

export const useGetUser = () => {
  const userId = auth?.currentUser?.uid
  return useQuery({
    enabled: !!userId,
    queryFn: getUser,
    queryKey: ['userProfile', userId],
  })
}

export const useUpdateUser = () => {
  const queryClient = useQueryClient()
  const userId = auth?.currentUser?.uid

  return useMutation<any, Error, ProfileData, { previousProfile: any }>({
    mutationFn: (profileData: ProfileData) => updateUser(profileData),
    onError: (error, newProfileData, context) => {
      queryClient.setQueryData(['userProfile', userId], context?.previousProfile)
      Alert.alert(getLocale('error'), getLocale('profileUpdateFailed'), [
        {
          text: getLocale('ok'),
        },
      ])
    },
    onMutate: async newProfileData => {
      await queryClient.cancelQueries({ queryKey: ['userProfile', userId] })
      const previousProfile = queryClient.getQueryData(['userProfile', userId])
      queryClient.setQueryData(['userProfile', userId], (oldData: any) => ({
        ...oldData,
        ...newProfileData,
        displayName: `${newProfileData?.firstName} ${newProfileData?.lastName}`,
      }))
      return { previousProfile }
    },
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey: ['userProfile', userId] })
    },
    onSuccess: () => {
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('profileUpdatedSuccessfully'),
        type: 'success',
      })
      router.back()
    },
  })
}

export const useChangePassword = () =>
  useMutation({
    mutationFn: (credentials: ChangePasswordCredentials) => changeUserPassword(credentials),
    onError: () => {
      Alert.alert(getLocale('error'), getLocale('passwordChangeFailed'), [
        {
          text: getLocale('ok'),
        },
      ])
    },
    onSuccess: () => {
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('passwordChangedSuccessfully'),
        type: 'success',
      })
      router.back()
    },
  })

export const useAddPassword = () =>
  useMutation({
    mutationFn: (credentials: AddPasswordCredentials) => addUserPassword(credentials),
    onError: () => {
      Alert.alert(getLocale('error'), getLocale('passwordAddFailed'), [
        {
          text: getLocale('ok'),
        },
      ])
    },
    onSuccess: () => {
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('passwordAddedSuccessfully'),
        type: 'success',
      })
      router.back()
    },
  })

export const useDeleteUser = () => {
  const queryClient = useQueryClient()

  return useMutation({
    mutationFn: deleteUserAccount,
    onError: () => {
      Alert.alert(getLocale('error'), getLocale('accountDeleteFailed'), [
        {
          text: getLocale('ok'),
        },
      ])
    },
    onSuccess: () => {
      deleteUser()
      setIsAuthenticated(false)
      queryClient.clear()
      Toast.show({
        text1: getLocale('successful'),
        text2: getLocale('accountDeletedSuccessfully'),
        type: 'success',
      })
    },
  })
}

export const useRegisterDevice = () =>
  useMutation({
    mutationFn: registerDevice,
  })

export const useRegisterDeviceToUser = () =>
  useMutation({
    mutationFn: registerDeviceToUser,
  })
