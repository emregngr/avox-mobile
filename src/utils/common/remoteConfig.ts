import {
  fetchAndActivate,
  getAll,
  getRemoteConfig,
  getValue,
  setConfigSettings,
  setDefaults,
} from '@react-native-firebase/remote-config'

import { Logger } from '@/utils/common/logger'

const defaultConfigs: Readonly<Record<string, string | boolean>> = {
  IS_MAINTENANCE: false,
  MIN_VERSION_SUPPORT: '1.0.0',
}

type ConfigKey = keyof typeof defaultConfigs

const remoteConfig = getRemoteConfig()

const setFirebaseConfig = async (): Promise<boolean> => {
  try {
    await setDefaults(remoteConfig, defaultConfigs)
    await setConfigSettings(remoteConfig, {
      minimumFetchIntervalMillis: __DEV__ ? 300 : 3600000,
    })

    const activated = await fetchAndActivate(remoteConfig)

    const log = {
      activated,
      all: getAll(remoteConfig),
      default: defaultConfigs,
    }

    Logger.breadcrumb('REMOTE CONFIG', 'info', log)
    return activated
  } catch (error: any) {
    Logger.breadcrumb('[setFirebaseConfig] error', 'error', error)
    return false
  }
}

const getBooleanValue = (key: ConfigKey): boolean => {
  const remoteValue = getValue(remoteConfig, key)

  if (remoteValue && remoteValue.getSource() !== 'default') {
    return remoteValue.asBoolean()
  }

  return defaultConfigs[key] as boolean
}

const getStringValue = (key: ConfigKey): string => {
  const remoteValue = getValue(remoteConfig, key)

  if (remoteValue && remoteValue.getSource() !== 'default') {
    return remoteValue.asString()
  }

  return defaultConfigs[key] as string
}

export default {
  defaultConfigs,
  getBooleanValue,
  getStringValue,
  setFirebaseConfig,
}
