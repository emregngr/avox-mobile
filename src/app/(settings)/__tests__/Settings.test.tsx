import { act, fireEvent, render, waitFor } from '@testing-library/react-native'
import * as Linking from 'expo-linking'
import { router } from 'expo-router'
import * as StoreReview from 'expo-store-review'
import type { ReactNode } from 'react'
import React from 'react'
import { Alert, AppState } from 'react-native'
import { checkNotifications, openSettings } from 'react-native-permissions'
import { SafeAreaProvider } from 'react-native-safe-area-context'

import Settings from '@/app/(settings)/settings'
import { useDeleteUser } from '@/hooks/services/useUser'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { Logger } from '@/utils/common/logger'
import { getStringValue } from '@/utils/common/remoteConfig'

const { mockedAuth } = require('@react-native-firebase/auth')

jest.mock('@/utils/common/logger')

const mockedLoggerBreadcrumb = Logger.breadcrumb as jest.MockedFunction<typeof Logger.breadcrumb>

jest.mock('@/locales/i18next')

const mockedGetLocale = getLocale as jest.MockedFunction<typeof getLocale>

jest.mock('@/store/locale')

const mockedUseLocaleStore = useLocaleStore as jest.MockedFunction<typeof useLocaleStore>

jest.mock('@/store/theme')

const mockedUseThemeStore = useThemeStore as jest.MockedFunction<typeof useThemeStore>

jest.mock('@/hooks/services/useUser')

const mockedUseDeleteUser = useDeleteUser as jest.MockedFunction<typeof useDeleteUser>

jest.mock('@/utils/common/remoteConfig')

const mockedGetStringValue = getStringValue as jest.MockedFunction<typeof getStringValue>

jest.mock('@/assets/icons/close', () => 'Close')

jest.mock('@/assets/icons/instagram.svg', () => 'Instagram')

jest.mock('@/assets/icons/tiktok.svg', () => 'Tiktok')

jest.mock('@/components/common', () => {
  const { View, Text, TouchableOpacity, Switch } = require('react-native')

  return {
    Header: ({
      title,
      rightIconOnPress,
      rightIcon,
    }: {
      title: string
      rightIconOnPress?: () => void
      rightIcon?: ReactNode
    }) => (
      <View testID="header">
        <Text testID="header-title">{title}</Text>
        {rightIconOnPress ? (
          <TouchableOpacity onPress={rightIconOnPress} testID="close-button">
            {rightIcon}
          </TouchableOpacity>
        ) : null}
      </View>
    ),

    ProfileItem: ({
      label,
      onPress,
      leftIcon,
      rightIcon = true,
      danger,
      testID,
      customLeftIcon,
    }: {
      label: string
      onPress?: () => void
      leftIcon?: string
      rightIcon?: boolean
      danger?: boolean
      testID?: string
      customLeftIcon?: ReactNode
    }) => (
      <TouchableOpacity onPress={onPress} style={{ opacity: danger ? 0.8 : 1 }} testID={testID}>
        <View>
          <Text testID="item-label">{label}</Text>
          {leftIcon ? <Text testID="item-left-icon">{leftIcon}</Text> : null}
          {customLeftIcon ? <Text testID="item-custom-left-icon">Custom</Text> : null}
          {rightIcon ? <Text testID="item-right-icon">chevron</Text> : null}
        </View>
      </TouchableOpacity>
    ),

    SafeLayout: ({ children }: { children: ReactNode }) => (
      <View testID="safe-layout">{children}</View>
    ),

    ThemedText: ({ children, ...props }: { children: string }) => (
      <Text {...props}>{children}</Text>
    ),

    Switch,
  }
})

const mockedCheckNotifications = checkNotifications as jest.MockedFunction<
  typeof checkNotifications
>

const mockedRemove = jest.fn()
let mockedAddEventListener: jest.SpyInstance
let mockedAlert: jest.SpyInstance

const renderWithSafeAreaProvider = (component: ReactNode) =>
  render(
    <SafeAreaProvider
      initialMetrics={{
        insets: { top: 0, left: 0, right: 0, bottom: 0 },
        frame: { x: 0, y: 0, width: 375, height: 812 },
      }}
    >
      {component}
    </SafeAreaProvider>,
  )

beforeEach(() => {
  mockedUseLocaleStore.mockReturnValue({ selectedLocale: 'en' })
  mockedUseThemeStore.mockReturnValue({ selectedTheme: 'light' })

  mockedUseDeleteUser.mockReturnValue({
    mutateAsync: jest.fn().mockResolvedValue(true),
  } as any)

  mockedAddEventListener = jest
    .spyOn(AppState, 'addEventListener')
    .mockReturnValue({ remove: mockedRemove })

  mockedAlert = jest.spyOn(Alert, 'alert').mockImplementation(() => {})

  mockedCheckNotifications.mockResolvedValue({ status: 'granted', settings: {} })

  mockedAuth.currentUser = { uid: 'test-user-id', email: 'test@example.com' }

  mockedGetStringValue.mockImplementation((key: string) => {
    if (key === 'TIKTOK_LINK') return 'https://tiktok.com/test'
    if (key === 'INSTAGRAM_LINK') return 'https://instagram.com/test'
    return ''
  })

  mockedGetLocale.mockImplementation((key: string) => key)
})

afterEach(() => {
  jest.restoreAllMocks()
})

describe('Settings Screen', () => {
  it('renders correctly', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
    expect(getByTestId('safe-layout')).toBeTruthy()
  })

  it('navigates back when close button is pressed', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
    fireEvent.press(getByTestId('close-button'))
    expect(router.back).toHaveBeenCalled()
  })

  it('calls delete user when confirmed', () => {
    const localMockMutateAsync = jest.fn().mockResolvedValue(true)
    mockedUseDeleteUser.mockReturnValue({ mutateAsync: localMockMutateAsync } as any)
    mockedAlert.mockImplementation((title, message, buttons) => {
      buttons?.find((b: any) => b.style === 'destructive')?.onPress?.()
    })
    const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
    fireEvent.press(getByTestId('delete-account-button'))
    expect(localMockMutateAsync).toHaveBeenCalled()
  })

  it('cleans up event listener on unmount', () => {
    const { unmount } = renderWithSafeAreaProvider(<Settings />)
    unmount()
    expect(mockedRemove).toHaveBeenCalled()
  })

  it('navigates to correct pages on item press', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
    fireEvent.press(getByTestId('settings-privacy-policy-button'))
    expect(router.navigate).toHaveBeenCalledWith('/settings-privacy-policy')
    fireEvent.press(getByTestId('settings-terms-of-use-button'))
    expect(router.navigate).toHaveBeenCalledWith('/settings-terms-of-use')
    fireEvent.press(getByTestId('faq-button'))
    expect(router.navigate).toHaveBeenCalledWith('/faq')
  })

  it('opens external social media links', () => {
    const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
    fireEvent.press(getByTestId('tiktok-button'))
    expect(Linking.openURL).toHaveBeenCalledWith('https://tiktok.com/test')
    fireEvent.press(getByTestId('instagram-button'))
    expect(Linking.openURL).toHaveBeenCalledWith('https://instagram.com/test')
  })

  it('opens app settings when notification switch is toggled', () => {
    const { getByRole } = renderWithSafeAreaProvider(<Settings />)
    const notificationSwitch = getByRole('switch')
    fireEvent(notificationSwitch, 'onValueChange', true)
    expect(openSettings).toHaveBeenCalled()
  })

  it('shows an alert if store review is not available', async () => {
    jest.spyOn(StoreReview, 'isAvailableAsync').mockResolvedValue(false)
    const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
    fireEvent.press(getByTestId('rate-app-button'))
    await act(async () => {})
    expect(mockedAlert).toHaveBeenCalled()
  })

  it('does not render delete account if user is not logged in', () => {
    mockedAuth.currentUser = null
    const { queryByTestId } = renderWithSafeAreaProvider(<Settings />)
    expect(queryByTestId('delete-account-button')).toBeNull()
  })

  it('refetches notification status when app becomes active', () => {
    renderWithSafeAreaProvider(<Settings />)
    expect(mockedCheckNotifications).toHaveBeenCalledTimes(1)
    const appStateChangeHandler = mockedAddEventListener.mock.calls[0][1]
    act(() => {
      appStateChangeHandler('active')
    })
    expect(mockedCheckNotifications).toHaveBeenCalledTimes(2)
  })

  describe('Error Handling', () => {
    it('logs error if remote config fails', async () => {
      mockedGetStringValue.mockImplementation(() => {
        throw new Error('Remote config failed')
      })
      renderWithSafeAreaProvider(<Settings />)
      await waitFor(() => {
        expect(mockedLoggerBreadcrumb).toHaveBeenCalledWith(
          'getTikTokLinkError',
          'error',
          expect.any(Error),
        )
        expect(mockedLoggerBreadcrumb).toHaveBeenCalledWith(
          'getInstagramLinkError',
          'error',
          expect.any(Error),
        )
      })
    })

    it('logs error if notification check fails', async () => {
      mockedCheckNotifications.mockRejectedValue(new Error('Permission check failed'))
      renderWithSafeAreaProvider(<Settings />)
      await waitFor(() => {
        expect(mockedLoggerBreadcrumb).toHaveBeenCalledWith(
          'checkNotificationPermissionError',
          'error',
          expect.any(Error),
        )
      })
    })

    it('logs error if store review request fails', async () => {
      jest.spyOn(StoreReview, 'isAvailableAsync').mockResolvedValue(true)
      jest.spyOn(StoreReview, 'requestReview').mockRejectedValue(new Error('Review failed'))
      const { getByTestId } = renderWithSafeAreaProvider(<Settings />)
      fireEvent.press(getByTestId('rate-app-button'))
      await waitFor(() => {
        expect(mockedLoggerBreadcrumb).toHaveBeenCalledWith(
          'requestReviewError',
          'error',
          expect.any(Error),
        )
      })
    })
  })
})

describe('Settings Screen Snapshot', () => {
  it('should render the Settings Screen successfully', () => {
    const { toJSON } = renderWithSafeAreaProvider(<Settings />)

    expect(toJSON()).toMatchSnapshot()
  })
})
