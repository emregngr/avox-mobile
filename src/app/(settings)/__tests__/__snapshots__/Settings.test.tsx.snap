// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Settings Screen Snapshot should render the Settings Screen successfully 1`] = `
<View
  testID="safe-layout"
>
  <View
    testID="header"
  >
    <Text
      testID="header-title"
    >
      settings
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="close-button"
    >
      <Close
        height={24}
        primaryColor="#D1D1D6"
        secondaryColor="#000000"
        width={24}
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerClassName="px-4"
    contentContainerStyle={
      {
        "paddingBottom": 20,
        "paddingTop": 64,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Text
        className="ml-4 mt-8 mb-2"
        color="text-70"
        type="body2"
      >
        general
      </Text>
      <View
        className="rounded-xl overflow-hidden bg-background-secondary"
      >
        <View
          className="h-14 flex-row justify-between items-center px-4 bg-background-secondary"
        >
          <View
            className="flex-row items-center"
          >
            <View
              color="#000000"
              name="bell-outline"
              size={24}
              testID="mocked-material-community-icon"
            />
            <Text
              className="ml-4"
              color="text-100"
              type="body1"
            >
              notification
            </Text>
          </View>
          <View
            className="justify-center"
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onTintColor="#B3D9FF"
              style={
                [
                  {
                    "alignSelf": "flex-start",
                  },
                  {
                    "backgroundColor": "#D1D1D6",
                    "borderRadius": 16,
                  },
                ]
              }
              thumbTintColor="#000000"
              tintColor="#D1D1D6"
              value={false}
            />
          </View>
        </View>
        <View
          className="border-b border-background-quaternary ml-14"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="rate-app-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              rateApp
            </Text>
            <Text
              testID="item-left-icon"
            >
              star-outline
            </Text>
            <Text
              testID="item-right-icon"
            >
              chevron
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="settings-privacy-policy-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              privacyPolicy
            </Text>
            <Text
              testID="item-left-icon"
            >
              file-lock-outline
            </Text>
            <Text
              testID="item-right-icon"
            >
              chevron
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="settings-terms-of-use-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              termsOfUse
            </Text>
            <Text
              testID="item-left-icon"
            >
              file-document-outline
            </Text>
            <Text
              testID="item-right-icon"
            >
              chevron
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="faq-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              faq
            </Text>
            <Text
              testID="item-left-icon"
            >
              help-circle-outline
            </Text>
            <Text
              testID="item-right-icon"
            >
              chevron
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 0.8,
            }
          }
          testID="delete-account-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              deleteAccount
            </Text>
            <Text
              testID="item-left-icon"
            >
              trash-can-outline
            </Text>
          </View>
        </View>
      </View>
      <Text
        className="ml-4 mt-8 mb-2"
        color="text-70"
        type="body2"
      >
        socialMedia
      </Text>
      <View
        className="rounded-xl overflow-hidden bg-background-secondary"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="tiktok-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              tikTok
            </Text>
            <Text
              testID="item-custom-left-icon"
            >
              Custom
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="instagram-button"
        >
          <View>
            <Text
              testID="item-label"
            >
              instagram
            </Text>
            <Text
              testID="item-custom-left-icon"
            >
              Custom
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
